### ISSUE TITLE
How can I reproduce the results shown in the paper?

### ISSUE THREAD

**Issue by fire537** [mat10d/EvolvePro#46]
Hello: 
I think this is an excellent project. I encountered some issues while trying to reproduce the results in the paper using the brenan dataset from DMS. In the first round, I randomly selected 16 mutation sites. For the following 9 rounds, I used the evolve_experimental() method to recommend 16 sites each time. However, the resulting curve I obtained was not as smooth and upward-trending as the one shown in the paper. Could you advise what I should do to better reproduce the results in the paper?

![Image](https://github.com/user-attachments/assets/93cbaff1-68f4-414d-a960-6c342f89aaf9)

**Comment by mat10d** [comment]
The random seed parameters that were used in the paper should be available in the DMS approach, and this will greatly affect your results, depending on what initial variants you begin with.

**Comment by chengaaaaasacs** [comment]
> 论文中使用的随机种子参数应该在 DMS 方法中可用，这将极大地影响您的结果，具体取决于您开始时的初始变体。

I encountered a similar issue while trying to reproduce the results using the evolve_experimental() method, as described in the paper.

Specifically, I applied the same pipeline to the Zika virus envelope protein. In Round 1, I randomly selected 16 variants. Then, over the next 9 rounds, I followed the same strategy (select top 16 predicted variants from df_sorted_all.csv, avoid duplicates, and match with experimental data). However, after 10 rounds and 160 total variants, only about 10 showed higher activity than wild-type.

I noticed in the thread you mentioned that DMS random seed choice can strongly affect the results, especially depending on the initial round. But I'm curious: DMS only affects the initial variant selection — how does that influence downstream predictions in later rounds?

If the choice of initial variants is indeed critical to reproducing the upward fitness trend shown in the paper, do you have any suggestions or guidelines on how to select them more effectively for different proteins like Zika?

**Comment by mat10d** [comment]
The main issue here is that when we ran EvolvePro on the DMS datasets (shown in the various `dms` subfolders of the repo) we ran simulations with multiple different starting variants, and reported the average of these simulations. `evolve_experimental` assumes that you are actually running EvolvePro with wet lab results, and therefore, you cannot afford multiple side-by-side runs of EvolvePro.

In terms of what the optimal initial variants are, that is a great question. As we postulated in the paper, it appears that a random forest regressor is 1) optimal in a low-n, high-p setting, so perfect for this setting and 2) that it iteratively improves when you stretch the distribution of training data with higher and higher activity variants. This is why we see that in the active learning loop, selecting the top_n variants round by round is better than any other selection method in improving activity round by round. Therefore, starting with a set of variants that work fairly well should assist you in improving more quickly, because your top layer will be better primed to suggest iterative improvements. That being said, such a situation is not easily achieved in a setting where you just have a WT protein that you'd like to improve in activity. We did test multiple other applicable first rounds, for example, seeking to optimize diversity (based on PLM embedding), but we didn't find anything that was significantly better than just selecting a random set of variants. 

My apologies for the delay in getting back to you!
